warnings when building 5.1.0 with gcc 7 (sid)

can be reproduced using kamailio/pkg-kamailio-docker:dev-sid image
```
CC (gcc) [M sms.so]		libsms_modem.o
libsms_modem.c: In function 'initmodem':
libsms_modem.c:230:36: warning: '%s' directive writing up to 128 bytes into a region of size 91 [-Wformat-overflow=]
    clen=sprintf(command,"AT+CPIN=\"%s\"\r",mdm->pin);
                                    ^~
libsms_modem.c:230:8: note: 'sprintf' output between 12 and 140 bytes into a destination of size 100
    clen=sprintf(command,"AT+CPIN=\"%s\"\r",mdm->pin);
    ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
```
CC (gcc) [M uac.so]		uac_reg.o
uac_reg.c: In function 'reg_ht_add':
uac_reg.c:558:81: warning: ?: using integer constants in boolean context [-Wint-in-bool-context]
uac_reg.c:502:13:
    p = p + ((bsize)?(bsize):(dst)->len) + 1; \
             ~~~~~~~                                                              
uac_reg.c:558:81:
  reg_copy_shm(&nr->callid, &str_empty, reg_keep_callid ? UAC_REG_TM_CALLID_SIZE : 0);
uac_reg.c:502:14: note: in definition of macro 'reg_copy_shm'
    p = p + ((bsize)?(bsize):(dst)->len) + 1; \
              ^~~~~
```
```
CC (gcc) [M auth_radius.so]		sterman.o
sterman.c: In function 'extract_avp':
sterman.c:78:61: warning: comparison between pointer and zero character constant [-Wpointer-compare]
   if(!q && p == vp->strvalue && vp->strvalue + sizeof(char) != '\0') {
                                                             ^~
sterman.c:78:33: note: did you mean to dereference the pointer?
   if(!q && p == vp->strvalue && vp->strvalue + sizeof(char) != '\0') {
                                 ^
```
```
CC (gcc) [M xmpp.so]		xode.o
xode.c: In function 'xode_get_tag':
xode.c:346:77: warning: comparison between pointer and zero character constant [-Wpointer-compare]
     if(parent == NULL || parent->firstchild == NULL || name == NULL || name == '\0') return NULL;
                                                                             ^~
xode.c:346:72: note: did you mean to dereference the pointer?
     if(parent == NULL || parent->firstchild == NULL || name == NULL || name == '\0') return NULL;
                                                                        ^
```
```
CC (gcc) [M xmpp.so]		xstream.o
xstream.c: In function '_xode_put_expatattribs':
xstream.c:34:20: warning: comparison between pointer and zero character constant [-Wpointer-compare]
     while (atts[i] != '\0')
                    ^~
xstream.c:34:12: note: did you mean to dereference the pointer?
     while (atts[i] != '\0')
            ^
CC (gcc) [M xmpp.so]		xode_from.o
xode_from.c: In function '_xode_put_expatattribs':
xode_from.c:35:20: warning: comparison between pointer and zero character constant [-Wpointer-compare]
     while (atts[i] != '\0')
                    ^~
xode_from.c:35:12: note: did you mean to dereference the pointer?
     while (atts[i] != '\0')
            ^
```
```
CC (gcc) [M ims_registrar_pcscf.so]		sec_agree.o
sec_agree.c: In function 'free_security_t':
sec_agree.c:216:5: warning: enumeration value 'SECURITY_NONE' not handled in switch [-Wswitch]
     switch (params->type)
     ^~~~~~
```
```
CC (gcc) [M db_mongodb.so]		mongodb_dbase.o
mongodb_dbase.c: In function 'db_mongodb_query':
mongodb_dbase.c:922:2: warning: 'mongoc_collection_find' is deprecated: Use mongoc_collection_find_with_opts instead [-Wdeprecated-declarations]
  mgcon->cursor = mongoc_collection_find (mgcon->collection,
  ^~~~~
In file included from /usr/include/libmongoc-1.0/mongoc-client.h:28:0,
                 from /usr/include/libmongoc-1.0/mongoc.h:28,
                 from mongodb_connection.h:26,
                 from mongodb_dbase.c:27:
/usr/include/libmongoc-1.0/mongoc-collection.h:145:1: note: declared here
 mongoc_collection_find (mongoc_collection_t *collection,
 ^~~~~~~~~~~~~~~~~~~~~~
CC (gcc) [M db_mongodb.so]		mongodb_connection.o
CC (gcc) [M db_mongodb.so]		db_mongodb_mod.o
make[4]: 'libsrdb1.so.1.0' is up to date.
LD (gcc) [M db_mongodb.so]		db_mongodb.so
CC (gcc) [M ndb_mongodb.so]		mongodb_client.o
mongodb_client.c: In function 'mongodbc_exec_cmd':
mongodb_client.c:290:4: warning: 'mongoc_collection_find' is deprecated: Use mongoc_collection_find_with_opts instead [-Wdeprecated-declarations]
    rpl->cursor = mongoc_collection_find (rpl->collection,
    ^~~
In file included from /usr/include/libmongoc-1.0/mongoc-client.h:28:0,
                 from /usr/include/libmongoc-1.0/mongoc.h:28,
                 from mongodb_client.h:25,
                 from mongodb_client.c:34:
/usr/include/libmongoc-1.0/mongoc-collection.h:145:1: note: declared here
 mongoc_collection_find (mongoc_collection_t *collection,
 ^~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/modules/auth_radius/sterman.c
+++ b/src/modules/auth_radius/sterman.c
@@ -75,7 +75,7 @@ static inline int extract_avp(
 	if(vp->attribute == attrs[A_SIP_AVP].v && vp->type == PW_TYPE_STRING) {
 		p = strchr(vp->strvalue, '#');
 		q = strchr(vp->strvalue, ':');
-		if(!q && p == vp->strvalue && vp->strvalue + sizeof(char) != '\0') {
+		if(!q && p == vp->strvalue && *(vp->strvalue + sizeof(char)) != '\0') {
 			r = p;
 			r = strchr(r++, '#');
 		} else

--- a/src/modules/db_mongodb/mongodb_dbase.c
+++ b/src/modules/db_mongodb/mongodb_dbase.c
@@ -919,10 +919,14 @@ int db_mongodb_query(const db1_con_t* _h, const db_key_t* _k, const db_op_t* _op
 		}
 		mgcon->nrcols = _nc;
 	}
+	#if MONGOC_CHECK_VERSION(1, 5, 0)
+	mgcon->cursor = mongoc_collection_find_with_opts (mgcon->collection,
+						seldoc, mgcon->colsdoc, NULL);
+	#else
 	mgcon->cursor = mongoc_collection_find (mgcon->collection,
 						MONGOC_QUERY_NONE, 0, 0, 0,
 						seldoc, mgcon->colsdoc, NULL);
-
+	#endif
 	if(!_r) {
 		goto done;
 	}

--- a/src/modules/sms/libsms_modem.c
+++ b/src/modules/sms/libsms_modem.c
@@ -213,7 +213,7 @@ int setmodemparams( struct modem *mdm )
 
 int initmodem(struct modem *mdm, cds_report cds_report_f)
 {
-	char command[100];
+	char command[MAX_CHAR_BUF+12];
 	char answer[100];
 	int retries=0;
 	int success=0;

--- a/src/modules/uac/uac_reg.c
+++ b/src/modules/uac/uac_reg.c
@@ -499,7 +499,7 @@ int reg_ht_add_byuser(reg_uac_t *reg)
 			strncpy((dst)->s, (src)->s, (src)->len); \
 			(dst)->len = (src)->len; \
 			(dst)->s[(dst)->len] = '\0'; \
-			p = p + ((bsize)?(bsize):(dst)->len) + 1; \
+			p = p + ((bsize)!=0?(bsize):(dst)->len) + 1; \
 		} \
 	} while(0);
 

--- a/src/modules/xmpp/xode.c
+++ b/src/modules/xmpp/xode.c
@@ -343,7 +343,7 @@ xode xode_get_tag(xode parent, const char* name)
     char *str, *slash, *qmark, *equals;
     xode step, ret;
 
-    if(parent == NULL || parent->firstchild == NULL || name == NULL || name == '\0') return NULL;
+    if(parent == NULL || parent->firstchild == NULL || name == NULL || *name == '\0') return NULL;
 
     if(strstr(name, "/") == NULL && strstr(name,"?") == NULL)
         return _xode_search(parent->firstchild, name, XODE_TYPE_TAG);

--- a/src/modules/xmpp/xode_from.c
+++ b/src/modules/xmpp/xode_from.c
@@ -32,7 +32,7 @@ static void _xode_put_expatattribs(xode current, const char **atts)
 {
     int i = 0;
     if (atts == NULL) return;
-    while (atts[i] != '\0')
+    while (*(atts[i]) != '\0')
     {
         xode_put_attrib(current, atts[i], atts[i+1]);
         i += 2;

--- a/src/modules/xmpp/xstream.c
+++ b/src/modules/xmpp/xstream.c
@@ -31,7 +31,7 @@ static void _xode_put_expatattribs(xode owner, const char** atts)
 {
     int i = 0;
     if (atts == NULL) return;
-    while (atts[i] != '\0')
+    while (*(atts[i]) != '\0')
     {
         xode_put_attrib(owner, atts[i], atts[i+1]);
         i += 2;

